senders/msteams/msteams.go

```diff
@@ -0,0 +1,219 @@
+package msteams
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/russross/blackfriday/v2"
+)
+
+// Sender implements moira sender interface via MS Teams
+type Sender struct {
+	frontURI string
+	logger   moira.Logger
+	location *time.Location
+	client   *http.Client
+}
+
+// Init initialises settings required for full functionality
+func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+	sender.logger = logger
+	sender.location = location
+	sender.frontURI = senderSettings["front_uri"]
+	sender.client = &http.Client{
+		Timeout: time.Duration(30) * time.Second,
+	}
+	return nil
+}
+
+// SendEvents implements Sender interface Send
+func (sender *Sender) SendEvents(events moira.NotificationEvents, contact moira.ContactData, trigger moira.TriggerData, plot []byte, throttled bool) error {
+
+	err := sender.isValidWebhookURL(contact.Value)
+	if err != nil {
+		return fmt.Errorf("invalid msteams webhook url: %s", err.Error())
+	}
+
+	request, err := sender.buildRequest(events, contact, trigger, plot, throttled)
+
+	if err != nil {
+		return fmt.Errorf("failed to build request: %s", err.Error())
+	}
+
+	response, err := sender.client.Do(request)
+
+	if err != nil {
+		return fmt.Errorf("failed to perform request: %s", err.Error())
+	}
+	defer response.Body.Close()
+
+	// read the entire response as required by https://golang.org/pkg/net/http/#Client.Do
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("failed to decode response: %s", err.Error())
```

Comment:
I would like to see error wrapping using go 1.13 error values here and elsewhere  https://github.com/golang/go/wiki/ErrorValueFAQ